File: frontend/src/types/match.ts
export interface SectionMatch {
  title: string;
  similarity: number;
}

export interface MatchResult {
  overall_score: number;
  keyword_score: number;
  semantic_score: number;
  keyword_coverage: number;
  missing_skills: string[];
  matched_skills: string[];
  section_scores: SectionMatch[];
  jd_role?: string;
  jd_seniority?: string;
}

File: backend/app/schemas/match.py
from __future__ import annotations

from pydantic import BaseModel
from typing import List, Optional


class SectionMatch(BaseModel):
    title: str
    similarity: float


class MatchResult(BaseModel):
    overall_score: float
    keyword_score: float
    semantic_score: float
    keyword_coverage: float
    missing_skills: List[str]
    matched_skills: List[str]
    section_scores: List[SectionMatch]
    jd_role: Optional[str] = None
    jd_seniority: Optional[str] = None
